nirodents/workflows/brainextraction.py

```diff
@@ -252,123 +271,152 @@ def init_rodent_brain_extraction_wf(
-        ds_final_inu = pe.Node(DataSink(base_directory=str(output_dir.parent)),
-                               name="ds_final_inu")
-        ds_final_msk = pe.Node(DataSink(base_directory=str(output_dir.parent)),
-                               name="ds_final_msk")
+        ds_final_inu = pe.Node(
+            DerivativesDataSink(
+                base_directory=str(output_dir), desc="preproc", compress=True,
+            ), name="ds_final_inu", run_without_submitting=True
+        )
+        ds_final_msk = pe.Node(
+            DerivativesDataSink(
+                base_directory=str(output_dir), desc="brain", suffix="mask", compress=True,
+            ), name="ds_final_msk", run_without_submitting=True
+        )
+
+        # fmt: off
         wf.connect([
-            (outputnode, ds_final_inu, [
-                ("out_corrected", f"{output_dir.name}.@inu_corrected")]),
-            (outputnode, ds_final_msk, [
-                ("out_mask", f"{output_dir.name}.@brainmask")]),
+            (inputnode, ds_final_inu, [("in_files", "source_file")]),
+            (inputnode, ds_final_msk, [("in_files", "source_file")]),
+            (outputnode, ds_final_inu, [("out_corrected", "in_file")]),
+            (outputnode, ds_final_msk, [("out_mask", "in_file")]),
         ])
+        # fmt: on
 
         if interim_checkpoints:
-            ds_report = pe.Node(DataSink(base_directory=str(output_dir.parent)),
-                                name="ds_report")
+            ds_report = pe.Node(
+                DerivativesDataSink(
+                    base_directory=str(output_dir), desc="brain",
+                    suffix="mask", datatype="figures"
+                ), name="ds_report", run_without_submitting=True
+            )
+            # fmt: off
             wf.connect([
-                (final_report, ds_report, [
-                    ("out_report", f"{output_dir.name}.@report")]),
+                (inputnode, ds_report, [("in_files", "source_file")]),
+                (final_report, ds_report, [("out_report", "in_file")]),
             ])
+            # fmt: on
 
         if ants_affine_init and interim_checkpoints:
-            ds_report_init = pe.Node(DataSink(base_directory=str(output_dir.parent)),
-                                     name="ds_report_init")
+            ds_report_init = pe.Node(
+                DerivativesDataSink(
+                    base_directory=str(output_dir), desc="init",
+                    suffix="mask", datatype="figures"
+                ), name="ds_report_init", run_without_submitting=True
+            )
+            # fmt: off
             wf.connect([
-                (init_report, ds_report_init, [
-                    ("out_report", f"{output_dir.name}.@init_report")]),
+                (inputnode, ds_report_init, [("in_files", "source_file")]),
+                (init_report, ds_report_init, [("out_report", "in_file")]),
             ])
+            # fmt: on
```

Comment:
I'm not familiar with this. What is it doing?

Reply:
this is to tell `black` (a python formatting and linting tool) to be reenabled after a `# fmt: off` right above. Basically these two comments hide sections of the code to `black`.